Remotion/Data/DomainObjects/Infrastructure/Serialization/FlattenedSerializationInfo.cs

```diff
@@ -54,7 +54,9 @@ public void AddBoolValue (bool value)
     /// <remarks>Note: This will use T to decide whether to recurse into an IFlattenedSerializable or not.</remarks>
     public void AddValue<T> (T? value)
     {
-      if (typeof(IFlattenedSerializable).IsAssignableFrom(typeof(T)))
+      if (typeof(T) == typeof(Type))
+        _objectWriter.AddSimpleValue(((Type?)(object?)value)?.AssemblyQualifiedName);
```

Comment:
Something I realized while checking the code duplication for type / string conversion:

While we don't need the handle here, it would be a good idea to use a handle anyway to store the type string. )The flattenedserializable works the same way.) There's just a handful of type strings in the appdomain but potentially hundreds or thousands of occurances. by putting the type string into a handle, the serialization set is compressed.

Please update.

It should be enough to simply change AddSimpleValue to AddHandle  / AddNullableHandle.